### PR TITLE
Mobile barcode scanner framework

### PR DESCRIPTION
**Pull request by brendancheng20**
- set up isMobile redux field
- toggle allowed routes based on isMobile
- created barcodeScanner class
- ~might~ have camera functionality working, but need a way to test over https on my phone so wanted to push to dev. I do not believe any of my isMobile stuff affects the normal app

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -90,56 +96,93 @@ class App extends React.Component<AppProps> {
     );
   };
 
+  detectMob() {
+    const toMatch = [
+      /Android/i,
+      /webOS/i,
```
Comment (juliasaveliff) [inline src/App.tsx:102]: what's webOS? jw

File: src/components/mobile/barcodeScanner.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  //   IToastProps,
+  //   Position,
+  //   Toaster
+} from "@blueprintjs/core";
+import "@blueprintjs/core/lib/css/blueprint.css";
+// import axios from "axios";
+import * as React from "react";
+import { connect } from "react-redux";
+import { RouteComponentProps, withRouter } from "react-router";
+// import { API_ROOT } from "../../utils/api-config";
```
Comment (juliasaveliff) [inline src/components/mobile/barcodeScanner.tsx:16]: can you delete all these unused imports before merging?

File: src/App.tsx (hunk 1 above)
Comment (brendancheng20) [reply src/App.tsx:102]: Not sure tbh, that's just what one of the things stackoverflow told me to include in my regex for detecting mobile

File: djreact/settings.py [hunk 3]
```diff
@@ -11,11 +11,12 @@
...
-    "rack-city-staging.herokuapp.com",
-    "rack-city-prod.herokuapp.com",
-    "127.0.0.1",
-    "localhost",
+    'rack-city-dev.herokuapp.com',
+    'rack-city-staging.herokuapp.com',
+    'rack-city-prod.herokuapp.com',
+    '127.0.0.1',
+    'localhost',
+    '192.168.1.17',
```
[the remark this answers is not in the document]
Comment (brendancheng20) [reply djreact/settings.py:19]: Oop forgot to lint. Will do + add your IP

**Pull request merged**